From aaf476d75b2667f79f1f137f2bf7fd3209c4d5a6 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Thu, 9 Apr 2015 11:41:05 +0800
Subject: [PATCH 6/6] atomisp: WA: sleep 10ms when power up ISP on byt

The patch fix CTS regression on ECS28A
android.hardware.camera2.cts.CameraDeviceTest#testCameraDeviceAbort
Other platforms had no such issue.

Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-7911
Change-Id: If394d9d2987e5970e128929903ce19cb9505c382
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c      | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index 0b3e120..e2542da 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -443,6 +443,14 @@ int atomisp_mrfld_power_up(struct atomisp_device *isp)
 	if (IS_CHT)
 		punit_ddr_dvfs_enable(false);
 
+	/*
+	 * FIXME:WA for ECS28A, with this sleep, CTS
+	 * android.hardware.camera2.cts.CameraDeviceTest#testCameraDeviceAbort
+	 * PASS, no impact on other platforms
+	*/
+	if (IS_BYT)
+		msleep(10);
+
 	/* writing 0x0 to ISPSSPM0 bit[1:0] to power off the IUNIT */
 	reg_value = intel_mid_msgbus_read32(PUNIT_PORT, MRFLD_ISPSSPM0);
 	reg_value &= ~MRFLD_ISPSSPM0_ISPSSC_MASK;
-- 
1.9.1

